dotnet/src/SemanticKernel.Core/Data/TextSearch/VectorStoreTextSearch.cs

```diff
@@ -203,17 +203,16 @@ private async Task<VectorSearchResults<TRecord>> ExecuteVectorSearchAsync(string
             OldFilter = searchOptions.Filter?.FilterClauses is not null ? new VectorSearchFilter(searchOptions.Filter.FilterClauses) : null,
 #pragma warning restore CS0618 // VectorSearchFilter is obsolete
             Skip = searchOptions.Skip,
-            Top = searchOptions.Top,
         };
 
         if (this._vectorizedSearch is not null)
         {
             var vectorizedQuery = await this._textEmbeddingGeneration!.GenerateEmbeddingAsync(query, cancellationToken: cancellationToken).ConfigureAwait(false);
 
-            return await this._vectorizedSearch.VectorizedSearchAsync(vectorizedQuery, vectorSearchOptions, cancellationToken).ConfigureAwait(false);
+            return await this._vectorizedSearch.VectorizedSearchAsync(vectorizedQuery, searchOptions.Top, vectorSearchOptions, cancellationToken).ConfigureAwait(false);
```

Comment:
We should consider applying the same decision (make Top a mandatory parameter) for ITextSearch - the same logic applies (/cc @markwallace-microsoft)

Reply:
Great catch, but IMO we need a separate PR for that (it's used in plenty of spaces, I am not sure how to run all related tests to ensure I dont break anything).

Reply:
@adamsitnik absolutely, opened #11381.